configuration_test.py

```diff
@@ -0,0 +1,189 @@
+#!/usr/bin/python
+import sys
+import os
+import re
+import subprocess
+import pkg_resources
+import warnings
+
+def configuration_test(ARGUMENTS):
+    # Determines whether to print traceback messages
+    debug = ARGUMENTS.get('debug', False) 
+    if debug == False:
+        # Hide traceback for configuration test only
+        # http://stackoverflow.com/questions/27674602/hide-traceback-unless-a-debug-flag-is-set
+        sys.tracebacklimit = 0 
+
+    # Checks initial prerequisites
+    check_python()
+    check_lyx()
+    check_gitlfs()
+    check_yamls()
+
+    # Uncomment if using
+    #check_r() 
+    #check_gslab_metropolis() 
+
+    # Loads arguments and configurations
+    mode = ARGUMENTS.get('mode', 'develop') # Gets mode; defaults to 'develop'
+
+    # Check stata
+    sf = check_stata(ARGUMENTS)
+
+    # Checks mode/version
+    if not (mode in ['develop', 'cache']):
+        raise PrerequisiteError("Error: %s is not a defined mode" % mode)
+    
+    # Get return list
+    if mode == 'cache':
+        cache_dir    = os.path.expanduser(load_yaml_value("user-config.yaml", "cache"))
+        check_cache(cache_dir)
+        return_list = [mode, sf, cache_dir]
+    else:
+        return_list = [mode, sf, None]
+
+    # Restore default tracebacklimit and return values
+    sys.tracebacklimit = 0
+    return return_list
+
+def check_python():
+    if sys.version_info[0] != 2:
+        raise PrerequisiteError('Please use python 2')
+    check_python_packages()
+
+def check_python_packages():
+    try:
+        import gslab_scons
```

Comment:
I didn't have the correct version of `gslab_scons` installed when I first ran this branch, and I got the following error:
```
$ scons
scons: Reading SConscript files ...
Enter stata_executable value or None to search for defaults: None
AttributeError: 'module' object has no attribute 'get_stata_executable':
```
Perhaps we could check the version of `gslab_scons` so that this error would have been more helpful?